### ISSUE TITLE
assertEquals(..) between Oracle table and csv dataset fails for NUMBER(n,0) columns

### ISSUE THREAD

**Issue by imps-od** [JDBDT/jdbdt#71]
Thanks for the nice db testing library!

We try to use it for verifying end-to-end test outcomes against "expected" data from CSV files.

But `JDBDT.assertEquals(expectedData, actualData)` always fails, even though I was running the test against CSV files I had just created using `JDBDT.write(..)`.

I narrowed it down to Numeric column values, with column types like NUMBER(38,0) in Oracle.

For example the value 5 is represented as `BigDecimal("5")` in the dataset from Oracle and `BigDecimal("5.0")` in the rows from the CSV file. The CSV file itself contains the value as "5".

Because `BigDecimal("5")` and `BigDecimal("5.0")` [are not equal](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html#equals(java.lang.Object)) in Java, the rows are not found as matching by assertEquals.

The code looks roughly like this:

```
final Format csv = new Format().hasHeader().alwaysEscapeOutput().useReadConversions();

final Table table = JDBDT.table(tableName).columns(columnNames).build(db);
final DataSet actualData = JDBDT.takeSnapshot(table);
final DataSet expectedData = JDBDT.read(table, csv, csvFile);

JDBDT.assertEquals(expectedData, actualData);
```

Do you have any suggestion how I could work around this ?

**Comment by edrdo** [comment]
@impoldo1 `JDBDT.assertEquals()` conforms to the `equals()` implementation for column values, hence `BigDecimal("5.0")` is interpreted as different from `BigDecimal("5")` (it merely follows the BigDecimal convention). <strike>From what I understood, though, `5.0` is written to the CSV file by JDBDT in correspondence to `BigDecimal("5")` ? Can you confirm/check that this is the case (it should not happen in principle).</strike>

**Comment by edrdo** [comment]
@impoldo1 I believe I may have found and fixed the issue. Please test using the HEAD version and let me know.

**Comment by imps-od** [comment]
Thanks for the quick turn-around on this. Literally you fixed this over night for us Canadians! :)

I checked out the JDBDT repo, did a local `mvn install`, changed my jdbdt dependency to `1.4.1-SNAPSHOT` and ran the test that was failing. It passed, so it looks like this particular issue was fixed by your code change.

However, I am still getting unexpected assertEquals failures on other tables. The deltas are the whole dataset so the JDBDT assertion log output is relatively unhelpful. Let me test further and I will comment here again.
